download-basket/src/main/java/downloadbasket/helpers/OGCServices.java

```diff
@@ -0,0 +1,208 @@
+        } catch(Exception e){
+            e.printStackTrace();
+        }
+        return s.toString();
+    }
+
+    /**
+     * Get bbox
+     * @param bbox
+     * @return
+     * @throws JSONException
+     */
+    private static String getBbox(JSONObject bbox) throws JSONException{
+        return bbox.getString(PARAM_BBOX_LEFT) + ","
+                + bbox.getString(PARAM_BBOX_BOTTOM) + ","
+                + bbox.getString(PARAM_BBOX_RIGHT) + ","
+                + bbox.getString(PARAM_BBOX_TOP);
+    }
+
+    /**
+     * Get WFS Query layer plugin WFS request
+     * @param download download details
+     * @param addNameSpaces add namespaces?
+     * @param writeParam write param
+     *
+     * @return filter writer
+     */
+    public static String getPluginFilter(JSONObject download, boolean addNameSpaces, boolean writeParam) throws JSONException {
+        JSONArray identifiers = new JSONArray(download.getString(PARAM_IDENTIFIERS));
+        StringWriter s = new StringWriter();
+        StringWriter filter = new StringWriter();
+        String croppingNameSpace = PropertyUtil.get("hsy.wfs.cropping.namespace");
+
+        try {
+            if(identifiers.length()>1) {
+                if(!addNameSpaces) {
+                    filter.append("<ogc:Or>");
+                } else {
+                    filter.append("<Or xmlns:ogc=\"http://www.opengis.net/ogc\">");
+                }
+            }
+
+            String intersectStart = "<ogc:Intersects>";
+            if(identifiers.length()==1 && addNameSpaces) {
+                intersectStart = "<Intersects xmlns:ogc=\"http://www.opengis.net/ogc\">";
+            }
+            String intersectEnd = "</ogc:Intersects>";
+            if(identifiers.length()==1 && addNameSpaces) {
+                intersectEnd = "</Intersects>";
+            }
+
+            for(int id=0;id<identifiers.length();id++){
+                JSONObject identifier = identifiers.getJSONObject(id);
+                String layerName = Helpers.getLayerNameWithoutNameSpace(identifier.getString("layerName"));
+                String uniqueColumn = identifier.getString("uniqueColumn");
+                String uniqueValue = identifier.getString("uniqueValue");
+                String cropGeomColumn = identifier.getString("geometryName");
+                String filterColumnType = identifier.getString("geometryColumn");
+                filter.append(intersectStart);
+                filter.append("<ogc:PropertyName>"+cropGeomColumn+"</ogc:PropertyName>");
+                filter.append("<ogc:Function name=\"querySingle\">");
+                filter.append("<ogc:Literal>"+croppingNameSpace+":"+layerName+"</ogc:Literal>");
+
+                filter.append("<ogc:Literal>"+cropGeomColumn+"</ogc:Literal>");
+                if(filterColumnType.equals("STRING")) {
+                    filter.append("<ogc:Literal>"+uniqueColumn + " LIKE '" + uniqueValue.trim() +"%" + "'</ogc:Literal>");
```

Comment:
Is any of these passed from the client? uniqueValue might be one at least? Probably needs some cleaning so it can't be used as an attack vector. Using a library to build the xml would handle this for you.